packages/SwingSet/src/kernel/state/storageWrapper.js

```diff
@@ -100,6 +153,7 @@ export function buildCrankBuffer(
       assert.typeof(key, 'string');
       additions.delete(key);
       deletions.add(key);
+      // liveGeneration += 1n; // XXX can this be made to work? I fear not...
```

Comment:
Wait, how is it safe to have this commented out? If someone deletes a key that was only in the `additions` buffer, the merged iterator will return the wrong value. Or, ok, maybe the `!deletions.has()` check protects against that case, but.. ok if someone adds a key, deletes it, re-adds it with a different value.. ok maybe the generation bump in `add` protects against that. But, in general, wouldn't it be more conservative to invalidate any iterator when a deletion happens?

Reply:
This is something we probably should have a conversation about.  It turns out that an important use case for iteration is to delete all the keys that are in some range; having deletion increment the generation number caused code that did that to break.  The only ways I can think of to accommodate that use case are to (a) capture all the keys into a separate array while iterating and then delete them in a second pass, which blows up if the cardinality is large (as when deleting a big table) or (b) add another syscall entry point specifically for deleting a key range, which seems oddly specialized for a syscall.

Reply:
Note for the record: after discussing this f2f, @warner and I concluded that we can remove the generation checks from `crankBuffer` since the kernel syscall implementation of vatstore already provides the protections needed.  However, I will add a comment describing the risks and necessary precautions when using the `getKeys` iterator.